content.php

```diff
@@ -11,39 +11,60 @@
     
     if ( $imageAttributes ) {
         $articleStyle = 'style="background-image: url( ' . $imageAttributes[0] . ' )"';
+        
     }
 }
 
 ?>
-<article id="post-<?php the_ID(); ?>" <?php post_class(); ?> <?php echo $articleStyle; ?>>
+
+<?php
+if ( ! empty( $imageAttributes ) ) { ?>
+    <style>
+    .site-main article.post-<?php the_ID(); ?> {
+        background: linear-gradient(45deg, rgba(0,0,0,0.3) 0%,rgba(0,0,0,0.1) 48%,rgba(0,0,0,0) 100%), url( <?php echo $imageAttributes[0]; ?> );
+        background-size: cover;
+    }
+<?php } ?>
+
+</style>
```

Comment:
This `</style>` should be placed inside the `if` statement above since if `$imageAttributes` is empty, then there will be an invalid `</style>` doing nothing